### ISSUE TITLE
[warning] Exception in Unstable 'resolveAdducts' Code

### ISSUE THREAD

**Issue by Adafede** [boecker-lab/sirius#19]
Hi, 

Since you wrote such explicit message, here it is:

`SEVERE07:51:55 - Exception in Unstable 'resolveAdducts' Code. PLease report this problem. Using unmodified Tree!
`

Don't know what could have happened, tell me which details you need! :)

**Comment by marcus-ludwig** [comment]
Hi Adafede, 

could you provide the corresponding spectrum and parameters to reproduce this issue?

Best,
Marcus

**Comment by Adafede** [comment]
Hi,

Yes of course!

Here are the parameters:

```
sirius --input=/Users/rutza/Downloads/MR_data/mzxml/pos/201125_PMA_MR_40.mzXML --output=sirius/metacombomics_pos config --AdductSettings.detectable="[M-H]-,[M]-,[M+K-2H]-,[M+Cl]-,[M-H2O-H]-,[M+Na-2H]-,[M+CH2O2-H]-,[M+Br]-,[M+C2H4O2-H]-,[M+C2HF3O2-H]-,[M+C2H3N-H]-" --AdductSettings.enforced="[M-H]-,[M+Cl]-,[M-H2O-H]-,[M+Br]-" --AdductSettings.fallback="[M-H]-" --AlgorithmProfile=orbitrap --ForbidRecalibration=ALLOWED --FormulaResultRankingScore=AUTO --FormulaResultThreshold=true --FormulaSearchDB=ALL --FormulaSettings.detectable=S,Br,Cl,B,Se --FormulaSettings.enforced=C,H,N,O,P --FormulaSettings.fallback=S --IsotopeMs2Settings=IGNORE --IsotopeSettings.filter=True --IsotopeSettings.multiplier=1 --MedianNoiseIntensity=0.015 --MotifDbFile=none --ms1.absoluteIntensityError=0.02 --ms1.minimalIntensityToConsider=0.0001 --ms1.relativeIntensityError=0.08 --MS1MassDeviation.allowedMassDeviation="10.0 ppm" --MS1MassDeviation.massDifferenceDeviation="5.0 ppm" --MS1MassDeviation.standardMassDeviation="10.0 ppm" --MS2MassDeviation.allowedMassDeviation="10.0 ppm" --MS2MassDeviation.standardMassDeviation="10.0 ppm" --NoiseThresholdSettings.absoluteThreshold=10000 --NoiseThresholdSettings.basePeak=NOT_PRECURSOR --NoiseThresholdSettings.intensityThreshold=0.0001 --NoiseThresholdSettings.maximalNumberOfPeaks=60 --NumberOfCandidates=50 --NumberOfCandidatesPerIon=1 --NumberOfStructureCandidates=10000 --PossibleAdductSwitches="[M+Cl]-:[M-H]-" --RecomputeResults=False --StructurePredictors=CSI_FINGERID --StructureSearchDB=ALL --Timeout.secondsPerInstance=0 --Timeout.secondsPerTree=0 --WriteSummaries=True --ZodiacClusterCompounds=false --ZodiacEdgeFilterThresholds.minLocalCandidates=1 --ZodiacEdgeFilterThresholds.minLocalConnections=10 --ZodiacEdgeFilterThresholds.thresholdFilter=0.95 --ZodiacEpochs.burnInPeriod=2000 --ZodiacEpochs.iterations=20000 --ZodiacEpochs.numberOfMarkovChains=10 --ZodiacLibraryScoring.lambda=1000 --ZodiacLibraryScoring.minCosine=0.5 --ZodiacNumberOfConsideredCandidatesAt300Mz=10 --ZodiacNumberOfConsideredCandidatesAt800Mz=50 --ZodiacRatioOfConsideredCandidatesPerIonization=0.2 --ZodiacRunInTwoSteps=true F Z S C
```

I actually had this warning many times on 40 files, I attach one representative example here:

https://we.tl/t-PlUUS5ZjGN

Tell me if you need more!

Best

**Comment by marcus-ludwig** [comment]
Great, 
I'll have a look into it and let you know if I need additional info.

Best

**Comment by marcus-ludwig** [comment]
> Here are the parameters:
> 
> ```
> sirius --input=/Users/rutza/Downloads/MR_data/mzxml/pos/201125_PMA_MR_40.mzXML --output=sirius/metacombomics_pos config --AdductSettings.detectable="[M-H]-,[M]-,[M+K-2H]-,[M+Cl]-,[M-H2O-H]-,[M+Na-2H]-,[M+CH2O2-H]-,[M+Br]-,[M+C2H4O2-H]-,[M+C2HF3O2-H]-,[M+C2H3N-H]-" --AdductSettings.enforced="[M-H]-,[M+Cl]-,[M-H2O-H]-,[M+Br]-" --AdductSettings.fallback="[M-H]-" --AlgorithmProfile=orbitrap --ForbidRecalibration=ALLOWED --FormulaResultRankingScore=AUTO --FormulaResultThreshold=true --FormulaSearchDB=ALL --FormulaSettings.detectable=S,Br,Cl,B,Se --FormulaSettings.enforced=C,H,N,O,P --FormulaSettings.fallback=S --IsotopeMs2Settings=IGNORE --IsotopeSettings.filter=True --IsotopeSettings.multiplier=1 --MedianNoiseIntensity=0.015 --MotifDbFile=none --ms1.absoluteIntensityError=0.02 --ms1.minimalIntensityToConsider=0.0001 --ms1.relativeIntensityError=0.08 --MS1MassDeviation.allowedMassDeviation="10.0 ppm" --MS1MassDeviation.massDifferenceDeviation="5.0 ppm" --MS1MassDeviation.standardMassDeviation="10.0 ppm" --MS2MassDeviation.allowedMassDeviation="10.0 ppm" --MS2MassDeviation.standardMassDeviation="10.0 ppm" --NoiseThresholdSettings.absoluteThreshold=10000 --NoiseThresholdSettings.basePeak=NOT_PRECURSOR --NoiseThresholdSettings.intensityThreshold=0.0001 --NoiseThresholdSettings.maximalNumberOfPeaks=60 --NumberOfCandidates=50 --NumberOfCandidatesPerIon=1 --NumberOfStructureCandidates=10000 --PossibleAdductSwitches="[M+Cl]-:[M-H]-" --RecomputeResults=False --StructurePredictors=CSI_FINGERID --StructureSearchDB=ALL --Timeout.secondsPerInstance=0 --Timeout.secondsPerTree=0 --WriteSummaries=True --ZodiacClusterCompounds=false --ZodiacEdgeFilterThresholds.minLocalCandidates=1 --ZodiacEdgeFilterThresholds.minLocalConnections=10 --ZodiacEdgeFilterThresholds.thresholdFilter=0.95 --ZodiacEpochs.burnInPeriod=2000 --ZodiacEpochs.iterations=20000 --ZodiacEpochs.numberOfMarkovChains=10 --ZodiacLibraryScoring.lambda=1000 --ZodiacLibraryScoring.minCosine=0.5 --ZodiacNumberOfConsideredCandidatesAt300Mz=10 --ZodiacNumberOfConsideredCandidatesAt800Mz=50 --ZodiacRatioOfConsideredCandidatesPerIonization=0.2 --ZodiacRunInTwoSteps=true F Z S C
> ```

You are specifying adducts for negative ion mode, but your data is in positive mode. Are these the same parameters you used for this data? If so, this is likely related to the issue.

**Comment by Adafede** [comment]
My bad... sent you the wrong pair as I ran both pos and neg mode and mixed between example and parameters >.> 

Here is the correct one (with the file I gave you):

```
sirius --input=exampleFile --output=sirius/metacombomics_pos config --AdductSettings.detectable="[M+H-H2O]+,[M+H-H4O2]+,[M]+,[M+H]+,[M+NH4]+,[M+Na]+,[M+K]+,[M+Na2-H]+,[M+2K-H]+,[M+CH5O]+,[M+C2H4N]+,[M+C2H8N]+,[M+C2H7N2]+,[M+C3H9O]+,[M+C2H3N+Na]+,[M+C2H7OS]+" --AdductSettings.enforced="[M+H]+,[M+Na]+,[M+K]+" --AdductSettings.fallback="[M+H]+,[M+Na]+,[M+K]+" --AlgorithmProfile=orbitrap --ForbidRecalibration=ALLOWED --FormulaResultRankingScore=AUTO --FormulaResultThreshold=true --FormulaSearchDB=ALL --FormulaSettings.detectable=S,Br,Cl,B,Se --FormulaSettings.enforced=C,H,N,O,P --FormulaSettings.fallback=S --IsotopeMs2Settings=IGNORE --IsotopeSettings.filter=True --IsotopeSettings.multiplier=1 --MedianNoiseIntensity=0.015 --MotifDbFile=none --ms1.absoluteIntensityError=0.02 --ms1.minimalIntensityToConsider=0.0001 --ms1.relativeIntensityError=0.08 --MS1MassDeviation.allowedMassDeviation="10.0 ppm" --MS1MassDeviation.massDifferenceDeviation="5.0 ppm" --MS1MassDeviation.standardMassDeviation="10.0 ppm" --MS2MassDeviation.allowedMassDeviation="10.0 ppm" --MS2MassDeviation.standardMassDeviation="10.0 ppm" --NoiseThresholdSettings.absoluteThreshold=100000 --NoiseThresholdSettings.basePeak=NOT_PRECURSOR --NoiseThresholdSettings.intensityThreshold=0.0001 --NoiseThresholdSettings.maximalNumberOfPeaks=60 --NumberOfCandidates=50 --NumberOfCandidatesPerIon=1 --NumberOfStructureCandidates=10000 --PossibleAdductSwitches="[M+Na]+:[M+H]+,[M+K]+:[M+H]+" --RecomputeResults=False --StructurePredictors=CSI_FINGERID --StructureSearchDB=ALL --Timeout.secondsPerInstance=0 --Timeout.secondsPerTree=0 --WriteSummaries=True --ZodiacClusterCompounds=false --ZodiacEdgeFilterThresholds.minLocalCandidates=1 --ZodiacEdgeFilterThresholds.minLocalConnections=10 --ZodiacEdgeFilterThresholds.thresholdFilter=0.95 --ZodiacEpochs.burnInPeriod=2000 --ZodiacEpochs.iterations=20000 --ZodiacEpochs.numberOfMarkovChains=10 --ZodiacLibraryScoring.lambda=1000 --ZodiacLibraryScoring.minCosine=0.5 --ZodiacNumberOfConsideredCandidatesAt300Mz=10 --ZodiacNumberOfConsideredCandidatesAt800Mz=50 --ZodiacRatioOfConsideredCandidatesPerIonization=0.2 --ZodiacRunInTwoSteps=true F Z S C
```

Sorry again!

**Comment by marcus-ludwig** [comment]
I found a minor issue. I believe your computation results should still be all correct. We'll soon release a fix.

**Comment by Adafede** [comment]
Great, happy it led to somthing! :)

**Comment by mfleisch** [comment]
fixed with release 4.5.2
